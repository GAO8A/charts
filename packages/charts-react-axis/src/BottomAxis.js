import React, { PropTypes } from 'react';
import XTick from './XTick';

const pathStyle = {
  fill: 'none',
  strokeWidth: 1,
  shapeRendering: 'crispEdges',
};

const renderTickWith = (interval) => (_, i) => {
  const offset = interval * (i + 1);
  // TODO: Label
  // const label = x(offset);

  return (
    <XTick
      key={i}
      label="Label"
      offset={offset}
    />
  );
};

const BottomAxis = (props, context) => {
  const { tickCount } = props;
  const { chart } = context;

  const path = `M0,6V0H${chart.width}V6`;
  const interval = chart.width / tickCount;
  const transform = `translate(0,${chart.height})`;

  const renderTick = renderTickWith(interval);

  return (
    <g
      className="axis axis--x"
      fill="none"
      fontSize="10"
      fontFamily="sans-serif"
      textAnchor="middle"
      transform={transform}>
      <path className="domain" d={path} stroke="#000" style={pathStyle} />
      {Array.from(Array(tickCount - 1)).map(renderTick)}
    </g>
  );
};

BottomAxis.propTypes = {
  tickCount: PropTypes.number,
};

BottomAxis.contextTypes = {
  chart: PropTypes.object,
};

export default BottomAxis;
