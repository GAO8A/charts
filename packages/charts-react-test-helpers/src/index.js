export { default as ChartProvider } from './ChartProvider';
