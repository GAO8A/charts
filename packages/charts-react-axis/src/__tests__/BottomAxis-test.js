import React from 'react';
import renderer from 'react-test-renderer';
import { ChartProvider } from '@ibm-design/charts-react-test-helpers';
import BottomAxis from '../BottomAxis';

describe('BottomAxis Component', () => {
  it('should render', () => {
    const tree = renderer.create(
      <ChartProvider>
        <BottomAxis tickCount={5} />
      </ChartProvider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
