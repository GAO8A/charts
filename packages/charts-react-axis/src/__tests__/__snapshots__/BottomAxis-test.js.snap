exports[`BottomAxis Component should render 1`] = `
<g
  className="axis axis--x"
  fill="none"
  fontFamily="sans-serif"
  fontSize="10"
  textAnchor="middle"
  transform="translate(0,500)">
  <path
    className="domain"
    d="M0,6V0H500V6"
    stroke="#000"
    style={
      Object {
        "fill": "none",
        "shapeRendering": "crispEdges",
        "strokeWidth": 1,
      }
    } />
  <g
    className="tick"
    transform="translate(100,0)">
    <line
      stroke="#000"
      x1="0.5"
      x2="0.5"
      y2="6" />
    <text
      dy="0.71em"
      fill="#000"
      x="0.5"
      y="9">
      Label
    </text>
  </g>
  <g
    className="tick"
    transform="translate(200,0)">
    <line
      stroke="#000"
      x1="0.5"
      x2="0.5"
      y2="6" />
    <text
      dy="0.71em"
      fill="#000"
      x="0.5"
      y="9">
      Label
    </text>
  </g>
  <g
    className="tick"
    transform="translate(300,0)">
    <line
      stroke="#000"
      x1="0.5"
      x2="0.5"
      y2="6" />
    <text
      dy="0.71em"
      fill="#000"
      x="0.5"
      y="9">
      Label
    </text>
  </g>
  <g
    className="tick"
    transform="translate(400,0)">
    <line
      stroke="#000"
      x1="0.5"
      x2="0.5"
      y2="6" />
    <text
      dy="0.71em"
      fill="#000"
      x="0.5"
      y="9">
      Label
    </text>
  </g>
</g>
`;
